Packs/Whois/Integrations/Whois/Whois.py

```diff
@@ -7183,7 +7183,8 @@ def get_whois_raw(domain, server="", previous=None, rfc3490=True, never_cut=Fals
     # (3 tries led to errno.ECONNRESET)
     else:
         raise WhoisException('(104) Connection Reset By Peer')
-
+    if not response:
+        raise Exception(f"Got an empty response for requested domain {request_domain}")
```

Comment:
```suggestion
        raise Exception(f"Got an empty response for the requested domain {request_domain} from the server {target_server}. Please check your firewall or proxy settings.")
```
In addition:
1. let's put this in the try-except block and ensure this is executed only when the status code is bad.
2. make sure a **warning** entry is returned if the *with_error* param is not checked.